### ISSUE TITLE
Linking the review ticket in the published lesson?

### ISSUE THREAD

**Issue by arojascastro** [programminghistorian/jekyll#812]
I was wondering if we can do more to give credit reviewers and editors, and I remembered that on Wikipedia you can get access to the *Article* and the *talk*. One one hand, the *final* version; on the other, its discussion. For instance: https://en.wikipedia.org/wiki/2018_Gaza_border_protests

I thought that we may want to explore the possibility of linking each tutorial with its discussion on Github in order to gain transparency and acknowledge the importance of the reviewing process. So my questions are: do you think it is a good idea? Is it possible to do this by adding a metadata field in the YAML block?

**Comment by ianmilligan1** [comment]
I'm a lame duck as I'm leaving soon but speaking as an author while I don't mind having the record of peer review available for the truly dedicated and curious, I would prefer not to have it directly linked from the finished article.

**Comment by acrymble** [comment]
I hear both of you on this. I respect @ianmilligan1 's comment that the published version is for readers, not for authors.

I also agree with @arojascastro that it would be great to expose more information about individual lessons (eg, #787 conversation about altmetric). Could we think about a fairly discrete expandable icon or something near the top of a lesson for those interested in this type of thing, but not glaring for the typical reader?

I've been thinking also it would be great to see from a lesson page if there were translated versions of that lesson. And other things come to mind. If we can make it discrete and not take away from the reader experience, maybe we can explore this type of thing further?

**Comment by arojascastro** [comment]
Ok, let me propose a different approach. What about this?

- recording the number of the reviewing ticket in the YAML block as a kind of identifier so it fullfills our open and transparent aims - it is also nice for documentation purposes (I can't never recall the ticket number attached related to an specific tutorial)

OR / AND

- not linking to the ticket on the published lesson but displaying the ticket number - maybe not in the header of the page but in footnote next to the "See page history" for example

**Comment by mdlincoln** [comment]
If we do want to expose this, it doesn't make sense to _only_ put it in the metadata but not display it somewhere on the rendered HTML page.

I agree with the idea of the discreet approach. We could add a link to the editorial history at the bottom of the lesson page near the suggested citation. (for technical reasons, it would be awkward to place it literally in the footer next to the "page history" link - and anyway, the page history is fundamentally different from the review discussion since that happened in a separate repository)

If we want to proceed with this, I can code it up. But someone will need to go through each lesson and add metadata with the full URL to the review ticket or pull request. Any volunteers? It can't just be the ticket number from programminghistorian/ph-submissions, because older lessons were reviewed in pull requests directly on this repo.

**Comment by arojascastro** [comment]
Ok, so from a tecnical point of view, there are some difficulties. Maybe it is not a good idea to implement this suggestion. The link to the review would increase the transparency and credit but I am not sure it would translate into impact (in contrast to the display of number of views). On the other hand, I think Matthew is already working on linking EN to SPA lessons.

**Comment by acrymble** [comment]
We already record review tickets in the YAML. Is this a different suggestion?

**Comment by arojascastro** [comment]
Do we? Maybe in the translations we do not... I cannot find it. The suggestion was to display the number in the footer or add a link to the actual ticket, but as I said maybe it is not good idea because there are some difficulties and no clear benefit.

**Comment by acrymble** [comment]
<img width="787" alt="screen shot 2018-04-27 at 11 03 26" src="https://user-images.githubusercontent.com/646647/39357310-6dcf03e0-4a0a-11e8-8fa0-57fdc21aaec2.png">

**Comment by arojascastro** [comment]
```
title: |
    Uso de SPARQL para acceder a datos abiertos enlazados
authors:
- Matthew Lincoln
date: 2015-11-24
translation_date: 2017-05-20
editors:
- Fred Gibbs
reviewers:
- Patrick Murray-John
- Jason Heppler
- Will Hanley
- Fred Gibbs
translator:
- Nuria Rodríguez Ortega
translation-editor:
- Antonio Rojas Castro
translation-reviewer:
- Antonio Rojas Castro
- Juan Antonio Pastor Sánchez
layout: lesson
redirect_from: /es/lessons/graph-databases-and-SPARQL
difficulty: 2
activity: acquiring
topics: [lod]
abstract: "Esta lección explica por qué numerosas instituciones culturales están adoptando bases de datos orientadas a grafos y cómo los investigadores pueden acceder a estos datos a través de consultas realizadas en el lenguaje llamado SPARQL."
```
Uhmmm. Can't find it

**Comment by acrymble** [comment]
You obviously forgot to put it in. That's a different issue entirely. Maybe we should be numbering our YAML fields so it's more obvious when one is missing, or providing a template for editors to use so they don't forget?

This is also related to the problem of needing #681 Copy Editing.

**Comment by arojascastro** [comment]
Yes, I must have forgotten. Can we make it compulsory so TRAVIS alerts us? I do not think we have added that number in any of our translations 😮

**Comment by acrymble** [comment]
It's easy to forget. I'm sure you're not the only one.

**Comment by walshbr** [comment]
It's also easy to forget since the fields don't have to be in a particular order. Makes me check over multiple times myself.

**Comment by ianmilligan1** [comment]
Yep. In theory moving to the Pull Request -> Review system should move us to reviewing persnickety things like this, esp. as we get used to the new workflow. On my other projects we use that workflow to catch things that we might not set Travis or other automated service to catch.

**Comment by mdlincoln** [comment]
1. Indeed, I don't currently check for a ticket number in the Travis checks. We can do that for sure. But first we really need to not just put the number, but the full URL, because as I said above:

>It can't just be the ticket number from programminghistorian/ph-submissions, because older lessons were reviewed in pull requests directly on this repo.

**Comment by acrymble** [comment]
Actually I think the oldest lessons tended to be reviewed via email. Is it easiest just to do this as a moving-forward thing where only those lessons with a review ticket are included in this? The rest can have a note: Open Review was not used when this lesson was in submission. Or something like that?

**Comment by mdlincoln** [comment]
Sure, the very oldest lessons can be excluded from the check. With that exception though, I don't think there's any point in "half" doing this by only implementing this for future lessons. Either fix up the metadata for those that we can, or just stop doing it, otherwise we're going to have a strange patchwork of metadata standards.

**Comment by acrymble** [comment]
is there any way to get a quick list of which ones don't have it? Maybe I can help.

**Comment by mdlincoln** [comment]
@acrymble working on it as we speak :)

**Comment by mdlincoln** [comment]
~Here's a listing from Travis: https://travis-ci.org/programminghistorian/jekyll/builds/372131689~

See more recent build below: https://github.com/programminghistorian/jekyll/issues/812#issuecomment-385187898

**Comment by acrymble** [comment]
The following (english only) I can fix via pointing to submission site tickets:

 - [x] 3 /lessons/building-static-sites-with-jekyll-github-pages.md:
 - [x] 78 /lessons/correspondence-analysis-in-R.md:
 - [x] 14 /lessons/getting-started-with-github-desktop.md:
 - [x] 6 /lessons/installing-omeka.md:
 - [x] 17 /lessons/intro-to-augmented-reality-with-unity.md:
 - [x] 23 /lessons/json-and-jq.md:
 - [x] 19 /lessons/r-basics-with-tabular-data.md:
 - [x] 4 /lessons/sonification.md:
 - [x] 29 /lessons/text-mining-with-extracted-features.md:
 - [x] 11 /lessons/transforming-xml-with-xsl.md:

I only found two that were via pull request reviews only:

 - [x] PR61 /lessons/getting-started-with-markdown.md: (created ticket: 165).
 - [x] PR131 /lessons/graph-databases-and-SPARQL.md: (created ticket: 166).

Too Old: - value should be NULL

- [x] /lessons/applied-archival-downloading-with-wget.md:
- [x] In /lessons/automated-downloading-with-wget.md:
- [x] In /lessons/topic-modeling-and-mallet.md:
- [x] In /lessons/transliterating.md:
- [x] In /lessons/understanding-regular-expressions.md:
- [x] In /lessons/up-and-running-with-omeka.md:
- [x] In /lessons/vector-layers-qgis.md:
- [x] In /lessons/viewing-html-files.md:
- [x] In /lessons/windows-installation.md:
- [x] In /lessons/working-with-text-files.md:
- [x] In /lessons/working-with-web-pages.md:
- [x] In /lessons/intro-to-bash.md:
- [x] In /lessons/intro-to-beautiful-soup.md:
- [x] In /lessons/introduction-and-installation.md:
- [x] In /lessons/keywords-in-context-using-n-grams.md:
- [x] In /lessons/linux-installation.md:
- [x] In /lessons/mac-installation.md:
- [x] In /lessons/manipulating-strings-in-python.md:
- [x] In /lessons/naive-bayesian.md:
- [x] In /lessons/normalizing-data.md:
- [x] In /lessons/output-data-as-html-file.md:
- [x] In /lessons/output-keywords-in-context-in-html-file.md:
- [x] In /lessons/preserving-your-research-data.md:
- [x] In /lessons/qgis-layers.md:
- [x] In /lessons/research-data-with-unix.md:
- [x] In /lessons/sustainable-authorship-in-plain-text-using-pandoc-and-markdown.md:
- [x] In /lessons/installing-python-modules-pip.md:
- [x] In /lessons/counting-frequencies.md:
- [x] In /lessons/creating-an-omeka-exhibit.md:
- [x] In /lessons/creating-and-viewing-html-files-with-python.md:
- [x] In /lessons/creating-network-diagrams-from-historical-sources.md:
- [x] In /lessons/data-mining-the-internet-archive.md:
- [x] In /lessons/downloading-multiple-records-using-query-strings.md:
- [x] In /lessons/extracting-keywords.md:
- [x] In /lessons/from-html-to-list-of-words-1.md:
- [x] In /lessons/from-html-to-list-of-words-2.md:
- [x] In /lessons/generating-an-ordered-data-set-from-an-OCR-text-file.md:
- [x] In /lessons/googlemaps-googleearth.md:
- [x] In /lessons/cleaning-data-with-openrefine.md:
- [x] In /lessons/cleaning-ocrd-text-with-regular-expressions.md:
- [x] In /lessons/code-reuse-and-modularity.md:
- [x] In /lessons/corpus-analysis-with-antconc.md:

**Comment by mdlincoln** [comment]
While I think it's good to go ahead in fixing up the metadata in these the lessons, I still want to make sure we've reached some consensus about **displaying a link to that review ticket on the lesson page**. 

![screenshot-2018-4-27 reshaping json with jq](https://user-images.githubusercontent.com/4392604/39380468-81a121c4-4a13-11e8-861f-804bdf58dd59.png)

Are there any strong reactions to displaying this link in a similar font to the Suggested citation & Author info here, right before the footer of the page? If we're feeling ok about it by **May 7th** I'll move ahead on implementing it.

**Comment by acrymble** [comment]
I'd prefer something more discrete. And since it's metadata, I feel it should be in the metadata block at the top. LInking a 'peer review' icon makes sense to me.

<img width="1359" alt="example-peer-review" src="https://user-images.githubusercontent.com/646647/39392767-ad6753fa-4ab3-11e8-8c8d-e43a77475bbb.png">


Or even more discrete, just a (?) next to the 'Reviewers' label.

**Comment by mdlincoln** [comment]
n.b. @arojascastro @mariajoafana Here is the remaining list of lessons that need their review tickets 😬 

* In /es/lecciones/contar-frecuencias.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/creando-diagramas-de-redes-desde-fuentes-historicas.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/crear-y-ver-archivos-html-con-python.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/de-html-a-lista-de-palabras-1.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/de-html-a-lista-de-palabras-2.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/editar-audio-con-audacity.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/escritura-sostenible-usando-pandoc-y-markdown.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/instalacion-linux.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/instalacion-mac.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/instalacion-windows.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/instalar-modulos-python-pip.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/intro-a-google-maps-y-google-earth.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/introduccion-a-bash.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/introduccion-a-markdown.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/introduccion-control-versiones-github-desktop.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/introduccion-e-instalacion.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/limpieza-de-datos-con-OpenRefine.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/manipular-cadenas-de-caracteres-en-python.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/mineria-de-datos-en-internet-archive.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/normalizar-datos.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/palabras-clave-en-contexto-n-grams.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/poniendo-omeka-a-funcionar.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/preservar-datos-de-investigacion.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/reutilizacion-de-codigo-y-modularidad.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/salida-de-datos-como-archivo-html.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/salida-palabras-clave-contexto-ngrams.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/sparql-datos-abiertos-enlazados.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/trabajar-con-archivos-de-texto.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/trabajar-con-paginas-web.md:
  - [x] 'review-ticket' is missing.
* In /es/lecciones/ver-archivos-html.md:
  - [x] 'review-ticket' is missing.

**Comment by arojascastro** [comment]
Thank you @mdlincoln. I can add the ticket number to the lessons that I edited and you @mariajoafana  can add it to yours and if there are still missing we can split them. Is that ok? This will take me some days though.

**Comment by mdlincoln** [comment]
Thank you @arojascastro - take your time.

**Comment by arojascastro** [comment]
Done with all translations! After merging and publishing maybe @mariajoafana @vgayolrs @jenniferisasi @jamotilla could click on each link and make sure I pasted the right ticket.

**Comment by mdlincoln** [comment]
Please see the examples of rendering peer review links on lesson headers: https://github.com/programminghistorian/jekyll/pull/816#issuecomment-386817783

(And please leave any comments about them in the pull request - not here!)

**Comment by drjwbaker** [comment]
Loving this work. Great job all. Close the ticket?

**Comment by mdlincoln** [comment]
No @drjwbaker we can close the ticket when the PR is finished.

**Comment by mdlincoln** [comment]
Closed by #816
